### PR TITLE
[WIP] Awesome API Documentation

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref https://github.com/ipfs/js-ipfs/issues/615

Preview: https://ipfs.github.io/js-ipfs-api/

### CODE REVIEW

File: src/api/add.js [hunk 1]
```diff
@@ -5,6 +5,23 @@ const addToDagNodesTransform = require('../add-to-dagnode-transform')
...
+   *
+   * @alias add
+   * @method
+   * @param {(Buffer|Stream|Array<Buffer|Stream>)} files - The content to add.
+   * @param {function(Error, {hash: string})} [callback]
+   * @returns {Promise<{hash: string}>|undefined}
+   *
+   * @memberof Api#
+   *
+   * @example
```
Comment (VictorBjelkholm) [inline src/api/add.js]: As I understand our previous discussions, examples should make use of callbacks, not promises.

File: src/api/add.js [hunk 2]
```diff
@@ -5,6 +5,23 @@ const addToDagNodesTransform = require('../add-to-dagnode-transform')
 const promisify = require('promisify-es6')
 
 module.exports = (send) => {
+  /**
+   * Add content to IPFS.
+   *
+   * @alias add
+   * @method
+   * @param {(Buffer|Stream|Array<Buffer|Stream>)} files - The content to add.
+   * @param {function(Error, {hash: string})} [callback]
```
Comment (VictorBjelkholm) [inline src/api/add.js]: Don't we get the `file` in the callback rather than a String with the hash?

File: src/index.js [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
+ * const ipfs = ipfsAPI('localhost', '5001', {protocol: 'http'})
+ * // leaving out the arguments will default to these values
+ *
+ * // or connect with multiaddr
+ * const ipfs = ipfsAPI('/ip4/127.0.0.1/tcp/5001')
+ *
+ * // or using options
+ * const ipfs = ipfsAPI({host: 'localhost', port: '5001', procotol: 'http'})
+ */
+function Api (hostOrMultiaddr, port, opts) {
```
Comment (VictorBjelkholm) [inline src/index.js]: Why this changed to just `Api` from `IpfsAPI`? Having ipfs in the name helps when dealing with stacktraces

File: src/api/add.js (hunk 1 above)
Comment (dignifiedquire) [inline src/api/add.js]: feel free to change

File: src/index.js (hunk 3 above)
Comment (dignifiedquire) [inline src/index.js]: can't remember, feel free to change back

**Comment by alanshaw** [comment]
Closing as this PR is significantly out of date and unlikely to be completed.

**Pull request closed without merging**